src/Control/Newtype/Generic.hs

```diff
@@ -115,13 +122,34 @@ instance Newtype n (O n) => IsNewtype n
 class Newtype n o => HasUnderlying o n
 instance Newtype n o => HasUnderlying o n
 
--- Compat shim for GHC up to 8.2, which are lousy at symmetry
-coerce' :: Coercible a b => b -> a
-coerce' = coerce
+-- | Two types are @Similar@ if they are built from
+-- the same type constructor and the same kind arguments.
+--
+-- @Sum Int@ and @Sum Bool@ are @Similar@.
+--
+-- @Sum Int@ and @Product Int@ are not @Similar@
+-- because they are built from different type constructors.
+--
+-- @Const Int Char@ and @Const Int Maybe@ are not @Similar@
+-- because they have different kind arguments.
+class Similar (n :: k) (n' :: k)
+instance (Similar' n n', Similar' n' n) => Similar n n'
```

Comment:
I think this would be easier to read if the name `Similar` was easier to distinguish from the name for `Similar'`.

Reply:
Unfortunately, users will end up seeing `Similar'`, which is fundamentally an implementation detail, in error messages and GHCi `:t` responses. I don't know of any way to avoid this without causing other problems. So either I can document that internal thing or I can pick a name that's close enough for people to guess the meaning. Bleh!

Reply:
I'd appreciate a bit of documentation for `Similar'` in any case!